oxen/src/py_remote_repo.rs

```diff
@@ -123,4 +236,49 @@ impl PyRemoteRepo {
             _ => Err(PyValueError::new_err("could not get branch")),
         }
     }
+
+    fn get_commit(&self, commit_id: String) -> PyResult<PyCommit> {
+        let commit = pyo3_asyncio::tokio::get_runtime()
+            .block_on(async { api::remote::commits::get_by_id(&self.repo, &commit_id).await });
+        match commit {
+            Ok(Some(commit)) => Ok(PyCommit { commit }),
+            _ => Err(PyValueError::new_err("could not get commit id {commit_id}")),
+        }
+    }
+
+    fn create_branch(&self, new_name: String) -> PyResult<PyBranch> {
+        let branch = pyo3_asyncio::tokio::get_runtime().block_on(async {
+            api::remote::branches::create_from_or_get(&self.repo, &new_name, &self.revision).await
+        });
+
+        match branch {
+            Ok(branch) => Ok(PyBranch {
+                name: branch.name,
+                commit_id: branch.commit_id,
+            }),
+            _ => Err(PyValueError::new_err("Could not get or create branch")),
+        }
+    }
+    fn checkout(&mut self, revision: String) -> PyResult<()> {
+        // TODO: Flatten
```

Comment:
Agreed, feels like we can avoid nesting here

Reply:
Updated, lmk if you had something different in mind for the commit case!